component/core/src/main/java/org/exoplatform/social/core/manager/IdentityManagerImpl.java

```diff
@@ -264,23 +272,14 @@ private void detectChanges(Profile specificProfile) {
         list.add(Profile.UpdateType.ABOUT_ME);
       }
       if (UserProfileComparator.hasChanged(specificProfile,
-                                           existingProfile,
-                                           Profile.FIRST_NAME,
-                                           Profile.LAST_NAME,
-                                           Profile.EMAIL,
-                                           Profile.POSITION,
-                                           Profile.GENDER,
-                                           Profile.CONTACT_PHONES,
-                                           Profile.CONTACT_IMS,
-                                           Profile.CONTACT_URLS,
-                                           Profile.EXTERNAL,
-                                           Profile.LAST_LOGIN_TIME)) {
+              existingProfile,
+              getProfilePropertySettingsService().getPropertySettings().stream().map(ProfilePropertySetting::getPropertyName).toList())) {
```

Comment:
Add in `ProfilePropertySettingsService` a method `getPropertySettingNames` instead of repeating this stream expression in a lot of places.

```suggestion
              getProfilePropertySettingsService().getPropertySettingNames()) {
```

Reply:
Done